lib/src/connection.dart

```diff
@@ -185,11 +187,16 @@ class PostgreSQLConnection implements PostgreSQLExecutionContext {
         .timeout(new Duration(seconds: timeoutInSeconds), onTimeout: _timeout);
   }
 
+  Future _open() async {
+    if(!_hasConnectedPreviously)
```

Comment:
I think, we not must throw exception, because this method call in execute, quary.
User can call quary() and we start opening connection, after user can call yet quary() and we must add in queue and wait response.
At the second request we should not throw an exception.